### PR TITLE
Sci 1034 variant annotator updates

### PR DESCRIPTION
**Pull request by godotgildor**
#dxapp.json
- [x] Upgrade to 14.04
- [x] Add help and extension to input spec.
- [x] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [x] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [x] Use `regionalOptions` 
- [x] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [ ] Add subjob restartability when applicable.
- [x] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Setting `"openSource": true`. 
- [x] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [x] Use AssetDepend instead of execDepend
- [x] Add default instance type that works for a general use case.
- [x] add new input if applicable

#src/code.sh src/code.py
- [x] GATK tarball should be removed from the resource project (or make a new resource project), and the applet should require GATK tarball as a required input. This way, the users are enforced to provide their own tarball. This would avoid trouble from non-academic entities without GATK license to use your tarball.

# Readme.md
- [x] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [ ] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [x] Basic testing using given input in Box or expected data
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.

### CODE REVIEW

File: sequencing/variant_annotator/dxapp.json [hunk 1]
```diff
@@ -91,29 +120,30 @@
...
-  "access": {
-    "network": [
-      "*"
-    ],
-     "allProjects": "VIEW"
+  "regionalOptions": {
+    "aws:us-east-1": {"systemRequirements": {"*": {"instanceType": "mem1_ssd1_x8"}},
+                      "assetDepends": [{"id": "record-FJ1pP980FJfZFzqYG88B8Zj1"}]},
+    "azure:westus": {"systemRequirements": {"*": {"instanceType": "azure:mem1_ssd1_x8"}},
+                     "assetDepends": [{"id": "record-FJ1vJgQ99qgf9kQk264zVpKv"}]}
```
Comment (yihchii) [inline sequencing/variant_annotator/dxapp.json]: Please check to use the latest finalized assets when ready.

**Review by yihchii — APPROVED** [review]
LGTM!
Before merging, please wait fand update `dxapp.json` accordingly to use the latest/finalized assets once they are ready.

File: sequencing/variant_annotator/Readme.md [hunk 2]
```diff
@@ -1,21 +1,28 @@
-<!-- dx-header -->
-# VCF QC (DNAnexus Platform App)
+# GATK3's VariantAnnotator
```
Comment (yihchii) [inline sequencing/variant_annotator/Readme.md]: "GATK3's VariantAnnotator" -> "GATK3 VariantAnnotator"

File: sequencing/variant_annotator/dxapp.json [hunk 3]
```diff
@@ -91,29 +120,30 @@
...
   },
-  "developers": ["user-tnperson", "user-auv13"],
   "openSource" : true,
-  "authorizedUsers" : ["org-geisinger_affiliate", "org-geisinger", "org-support", "user-codushlaine", "user-semanti.mukherjee", "user-alexander.li", "user-shannon.bruse", "user-jan.freudenberg", "user-crisvh"]
+  "details": {
+    "upstreamUrl": "https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.php",
+    "whatsNew": "* 1.0.0 First version of the VariantAnnotator app."
```
Comment (yihchii) [inline sequencing/variant_annotator/dxapp.json]: "* 1.0.0 First version of the VariantAnnotator app." -> "* 1.0.0 First version of the GATK3 VariantAnnotator app."

File: sequencing/variant_annotator/dxapp.json (hunk 1 above)
Comment (yihchii) [reply sequencing/variant_annotator/dxapp.json]: Please update record IDs to:
aws:us-east-1: record-FJ3zf8j08zVZFzqYG88Pbk7z
azure:westus: record-FJ4058j9Vf9ZBBVB2Q7b3VQz

**Review by yihchii — COMMENTED** [review]
Added some miner suggestions.

**Comment by yihchii** [comment]
@slzarate , Please help review Readme.md!

File: sequencing/variant_annotator/Readme.md [hunk 4]
```diff
@@ -1,21 +1,28 @@
...
 
-Calls [GATK VariantAnnotator](https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.php)
 
-This is the source code for an app that runs on the DNAnexus Platform.
-For more information about how to run or modify it, see
-https://wiki.dnanexus.com/.
-<!-- /dx-header -->
+## What does this app do?
 
+This app run's the [GATK VariantAnnotator](https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.php).
```
Comment (slzarate) [inline sequencing/variant_annotator/Readme.md]: runs

File: sequencing/variant_annotator/Readme.md [hunk 5]
```diff
@@ -1,21 +1,28 @@
...
+This tool is meant to annotate the variant calls in a given `VCF` file based on their context.
```
Comment (slzarate) [inline sequencing/variant_annotator/Readme.md]: I don't know if VCF should be monotype `VCF`

File: sequencing/variant_annotator/Readme.md [hunk 6]
```diff
@@ -1,21 +1,28 @@
...
 
-This is the source code for an app that runs on the DNAnexus Platform.
-For more information about how to run or modify it, see
-https://wiki.dnanexus.com/.
-<!-- /dx-header -->
+## What does this app do?
 
+This app run's the [GATK VariantAnnotator](https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.php).
+This tool is meant to annotate the variant calls in a given `VCF` file based on their context.  
+Given an array of `VCF` files, this app will launch a separate sub-job for each `VCF` and run GATK's VariantAnnotator.  Users can request that
```
Comment (slzarate) [inline sequencing/variant_annotator/Readme.md]: I think the term is "subjob". Also, remove line break

File: sequencing/variant_annotator/Readme.md [hunk 7]
```diff
@@ -1,21 +1,28 @@
...
-Readme.developer.md.) Once you're done, you can remove these TODO
-comments.
+## What data are required for this app to run?
 
-For more info, see https://wiki.dnanexus.com/Developer-Portal.
--->
+This app requires:
+
+- An array of `VCF` files to be annotated.
+- A matching array of `TBI` index files for those `VCFs`.
```
Comment (slzarate) [inline sequencing/variant_annotator/Readme.md]: - A matching array of index files for those VCF files (`*.vcf.gz.tbi`).

File: sequencing/variant_annotator/Readme.md [hunk 8]
```diff
@@ -1,21 +1,28 @@
...
+- The GATK3 `JAR` file.
```
Comment (slzarate) [inline sequencing/variant_annotator/Readme.md]: - The GATK3 `*.jar` file.

File: sequencing/variant_annotator/Readme.md [hunk 9]
```diff
@@ -1,21 +1,28 @@
...
+- The reference genome the variants were called against in `FASTA` format.
```
Comment (slzarate) [inline sequencing/variant_annotator/Readme.md]: Should the "FASTA" be in `FASTA`?

File: sequencing/variant_annotator/dxapp.json (hunk 1 above)
Comment (godotgildor) [reply sequencing/variant_annotator/dxapp.json]: Done

File: sequencing/variant_annotator/Readme.md (hunk 2 above)
Comment (godotgildor) [reply sequencing/variant_annotator/Readme.md]: Done

File: sequencing/variant_annotator/dxapp.json (hunk 3 above)
Comment (godotgildor) [reply sequencing/variant_annotator/dxapp.json]: Done

File: sequencing/variant_annotator/Readme.md (hunk 4 above)
Comment (godotgildor) [reply sequencing/variant_annotator/Readme.md]: Done

File: sequencing/variant_annotator/Readme.md (hunk 6 above)
Comment (godotgildor) [reply sequencing/variant_annotator/Readme.md]: Done

File: sequencing/variant_annotator/Readme.md (hunk 7 above)
Comment (godotgildor) [reply sequencing/variant_annotator/Readme.md]: Done

File: sequencing/variant_annotator/Readme.md (hunk 8 above)
Comment (godotgildor) [reply sequencing/variant_annotator/Readme.md]: Done

**Comment by godotgildor** [comment]
Thanks @yihhwang !  Merging.

**Pull request merged**